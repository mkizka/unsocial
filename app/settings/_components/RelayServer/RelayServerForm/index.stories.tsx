import type { Meta, StoryObj } from "@storybook/react";

import { RelayServerForm } from ".";

const meta: Meta<typeof RelayServerForm> = {
  component: RelayServerForm,
  args: {
    onSubmit: async () => {
      await new Promise((resolve) => setTimeout(resolve, 1000));
      return {
        type: "success",
        message: "成功メッセージ",
      };
    },
    relayServers: [
      {
        inboxUrl: "https://relay.example.com/inbox",
        status: "ACCEPTED",
      },
      {
        inboxUrl: "https://relay2.example.com/inbox",
        status: "SENT",
      },
      {
        inboxUrl: "https://relay3.example.com/inbox",
        status: "FAILED",
      },
    ],
  },
};

export default meta;

type Story = StoryObj<typeof RelayServerForm>;

export const Default: Story = {};

export const Empty: Story = {
  args: {
    relayServers: [],
  },
};

export const Error: Story = {
  args: {
    onSubmit: async () => {
      await new Promise((resolve) => setTimeout(resolve, 1000));
      return {
        type: "error",
        message: "エラーメッセージ",
      };
    },
  },
};
