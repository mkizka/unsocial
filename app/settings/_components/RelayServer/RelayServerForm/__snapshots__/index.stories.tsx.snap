// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings/_components/RelayServer/RelayServerForm Default smoke-test 1`] = `
<div class="rounded-lg border bg-card p-4 text-card-foreground shadow-sm">
  <form action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')">
    <div class="space-y-2">
      <label class="block font-bold"
             for="inbox-url"
      >
        リレーサーバー
      </label>
      <input class="flex h-10 w-full rounded-md border border-input bg-input-inner px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
             id="inbox-url"
             required
             placeholder="https://example.com/inbox"
             value="https://example.com/inbox"
             name="inbox-url"
      >
      <div class="flex items-center">
        <button class="relative inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 ml-auto">
          <span class>
            変更する
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`settings/_components/RelayServer/RelayServerForm Empty smoke-test 1`] = `
<div class="rounded-lg border bg-card p-4 text-card-foreground shadow-sm">
  <form action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')">
    <div class="space-y-2">
      <label class="block font-bold"
             for="inbox-url"
      >
        リレーサーバー
      </label>
      <input class="flex h-10 w-full rounded-md border border-input bg-input-inner px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
             id="inbox-url"
             required
             placeholder="未設定"
             name="inbox-url"
      >
      <div class="flex items-center">
        <button class="relative inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 ml-auto">
          <span class>
            変更する
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`settings/_components/RelayServer/RelayServerForm Error smoke-test 1`] = `
<div class="rounded-lg border bg-card p-4 text-card-foreground shadow-sm">
  <form action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')">
    <div class="space-y-2">
      <label class="block font-bold"
             for="inbox-url"
      >
        リレーサーバー
      </label>
      <input class="flex h-10 w-full rounded-md border border-input bg-input-inner px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50"
             id="inbox-url"
             required
             placeholder="https://example.com/inbox"
             value="https://example.com/inbox"
             name="inbox-url"
      >
      <div class="flex items-center">
        <button class="relative inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 ml-auto">
          <span class>
            変更する
          </span>
        </button>
      </div>
    </div>
  </form>
</div>
`;
